network/debinterfaces/activate.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package debinterfaces
+
+import (
+	"fmt"
+	"os"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/juju/loggo"
+	"github.com/juju/utils/clock"
+	"github.com/pkg/errors"
+)
+
+var logger = loggo.GetLogger("juju.network.debinterfaces")
+
+// ActivationParams contains options to use when bridging interfaces
+type ActivationParams struct {
+	BackupFilename string
+	Clock          clock.Clock
+	// map deviceName -> bridgeName
+	Devices          map[string]string
+	DryRun           bool
+	Filename         string
+	ReconfigureDelay int
+	Timeout          time.Duration
+}
+
+// ActivationResult captures the result of actively bridging the
+// interfaces using ifup/ifdown.
+type ActivationResult struct {
+	Stdout []byte
+	Stderr []byte
+	Code   int
+}
+
+func activationCmd(oldContent, newContent string, params *ActivationParams) string {
+	if params.ReconfigureDelay < 0 {
+		params.ReconfigureDelay = 0
+	}
+	deviceNames := make([]string, len(params.Devices))
+	i := 0
+	for k := range params.Devices {
+		deviceNames[i] = k
+		i++
+	}
+	sort.Strings(deviceNames)
+	return fmt.Sprintf(`
+#!/bin/bash
+
+set -eu
+
+: ${DRYRUN:=}
+
+if [ $DRYRUN ] && [ %[2]d == 10 ]; then sleep %[2]d; fi
```

Comment:
so if we have a script, and we want something to try it, it is natural that they might use DRYRUN to test it, and it is also not particularly surprising to have a sleep time of 10.
IMO, if you're going to have a magic connection between test infrastructure and script running (you have to exactly set X and Y to values that will trigger explicit behavior) it should be clearly marked as magic values rather than "10" and DRYRUN is nonzero.
Those feel much more under the "regular values that might accidentally trigger the magic behavior".